gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/remote/DriverRemoteTraversalSideEffects.java

```diff
@@ -89,9 +100,8 @@ public DriverRemoteTraversalSideEffects(final Client client, final UUID serverSi
                 keys = client.submitAsync(msg).get().all().get().stream().map(r -> r.getString()).collect(Collectors.toSet());
             } catch (Exception ex) {
                 final Throwable root = ExceptionUtils.getRootCause(ex);
-                if (root.getMessage().equals("Could not find side-effects for " + serverSideEffect + "."))
-                    keys = Collections.emptySet();
-                else
+                final String exMsg = null == root ? "" : root.getMessage();
+                if (!exMsg.equals("Could not find side-effects for " + serverSideEffect + "."))
```

Comment:
I know you didn't add this, but I really hate that we had to do this message checking - need a JIRA ticket to improve this (or perhaps a comment on an existing ticket regarding exceptions coming back from the server if there's a relevant one like that).